### DISCUSSION TITLE
Finding out if all my formulae are Apple Silicon-ready

### DISCUSSION THREAD

**Discussion by adiabatic** [Homebrew/discussions#1583] [Everyday usage]
Like many people, I'm mulling over a possible upgrade to an Apple Silicon-based Mac in the next few months. However, I don't really understand whether I'll be able to run all my terminals as native arm64 programs or if I'll have to run them all in Rosetta so I can continue to run one or two utilities that haven't (yet) been verified to work with Apple Silicon.

Quite honestly, I'm kind of surprised that "How can/should I run all my brewed software on Apple Silicon?" isn't in [the FAQ](https://docs.brew.sh/FAQ).

In case it matters, I use these terminal apps:

- Terminal
- iTerm 2
- Visual Studio Code (it's got one built in, and it's rather nice)

I then ran this in the Terminal in an attempt to collect all the installed formulae I have that don't have a Big Sur bottle:

```fish
brew info --json=v1 --installed | jq 'map(select(.bottle?.stable.files.arm64_big_sur.url == null) | .name)'
```

This gave me the following:

```json
[
  "diff-so-fancy",
  "dungeon",
  "git-filter-repo",
  "heroku",
  "heroku-node",
  "hledger",
  "mysql@5.6",
  "pandoc",
  "ruby-build",
  "xorgproto"
]
```

I'm not sure if this is a complete list with no false positives or negatives, but it seems reasonable-ish. Feedback is very much welcome, especially since I'm very new to `jq`.

I can probably live without `dungeon`, but I'm definitely going to need `hledger` (a Haskell program) and that old version of MySQL for my day job in the future. Will I be stuck needing to use run-in-Rosetta Terminal and text-editor apps until I have arm64 bottles of all my formulae?

Thanks in advance!

**Comment by SMillerDev** [comment]
Haskell is pretty much the last programming language in brew that does not have a working port for arm on macOS so I think you might be stuck with Rosetta for a while. Some of these might also have bottle :unneeded though, in which case they will run natively.

**Comment by tjluoma** [comment]
FWIW - you can install `pandoc` from [the latest installer](https://github.com/jgm/pandoc/releases/latest) until it is M1 compatible. I've been using it and it works just fine. I also added <https://github.com/jgm/pandoc/releases/> to my RSS reader so I will know when there are new releases to install/update.
